lib/puppet/type/cisco_interface.rb

```diff
@@ -698,6 +724,206 @@ def is_to_s(value)
     end
   end # property stp_vlan_port_priority
 
+  ###########################
+  # private vlan attributes #
+  ###########################
+
+  newproperty(:switchport_mode_private_vlan_host) do
+    desc 'Switchport private host mode of the interface.'
+
+    newvalues(
+      :host,
+      :promiscuous,
+      :disabled)
+  end # property switchport_mode_private_vlan_host
+
+  newproperty(:switchport_mode_private_vlan_host_association, array_matching: :all) do
+    format = '["primary_vlan", "secondary_vlan"]'
+    desc "An array of #{format} pairs. "\
+         "Valid values match format #{format}. "\
+         'primary_vlan and secondary_vlan are integers.'
+    match_error = "must be of format #{format}. "\
+                  'primary_vlan and secondary_vlan must be specified as integers.'
+
+    validate do |value|
+      fail "Vlan '#{value}' #{match_error}" unless
+            value.kind_of? String
+      fail "Vlan '#{value}' #{match_error}" unless
```

Comment:
You need to handle the default case in the validate method:

Example: https://github.com/cisco/cisco-network-puppet-module/blob/af9ee0fb511d37fa97c75bae80297b0d5572633b/lib/puppet/type/cisco_evpn_vni.rb#L117